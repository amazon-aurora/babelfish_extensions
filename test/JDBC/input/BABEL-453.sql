use master;
go

CREATE TABLE t453_real(a real);
CREATE TABLE t453_dp(a double precision);

-- binary -> real
declare @a binary(4) = 0xabcdabcd;
select (cast (@a as real));
go
declare @a binary(4) = 0xabcdabcd;
insert into t453_real values (@a);
go

-- binary -> double precision
declare @a binary(4) = 0xabcdabcd;
select (cast (@a as double precision));
go
declare @a binary(4) = 0xabcdabcd;
insert into t453_dp values (@a);
go

-- varbinary -> real
declare @a varbinary(4) = 0xabcdabcd;
select (cast (@a as real));
go
declare @a varbinary(4) = 0xabcdabcd;
insert into t453_real values (@a);
go

-- varbinary -> double precision
declare @a varbinary(4) = 0xabcdabcd;
select (cast (@a as double precision));
go
declare @a varbinary(4) = 0xabcdabcd;
insert into t453_dp values (@a);
go

-- reported case
CREATE PROCEDURE p453(@val real) AS
BEGIN
  DECLARE @BinaryVariable sys.varbinary(4) = @val
  PRINT @BinaryVariable
  PRINT cast(@BinaryVariable as real)
END
GO
EXEC p453 1.1;
GO

DROP PROCEDURE p453;
DROP TABLE t453_real;
DROP TABLE t453_dp;
GO
