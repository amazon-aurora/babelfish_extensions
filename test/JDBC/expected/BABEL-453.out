use master;
go


CREATE TABLE t453_real(a real);
CREATE TABLE t453_dp(a double precision);
-- binary -> real
declare @a binary(4) = 0xabcdabcd;
select (cast (@a as real));
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: cannot cast type "binary" to real)~~

declare @a binary(4) = 0xabcdabcd;
insert into t453_real values (@a);
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: column "a" is of type real but expression is of type "binary")~~


-- binary -> double precision
declare @a binary(4) = 0xabcdabcd;
select (cast (@a as double precision));
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: cannot cast type "binary" to double precision)~~

declare @a binary(4) = 0xabcdabcd;
insert into t453_dp values (@a);
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: column "a" is of type double precision but expression is of type "binary")~~


-- varbinary -> real
declare @a varbinary(4) = 0xabcdabcd;
select (cast (@a as real));
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: cannot cast type varbinary to real)~~

declare @a varbinary(4) = 0xabcdabcd;
insert into t453_real values (@a);
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: column "a" is of type real but expression is of type varbinary)~~


-- varbinary -> double precision
declare @a varbinary(4) = 0xabcdabcd;
select (cast (@a as double precision));
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: cannot cast type varbinary to double precision)~~

declare @a varbinary(4) = 0xabcdabcd;
insert into t453_dp values (@a);
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: column "a" is of type double precision but expression is of type varbinary)~~


-- reported case
CREATE PROCEDURE p453(@val real) AS
BEGIN
  DECLARE @BinaryVariable sys.varbinary(4) = @val
  PRINT @BinaryVariable
  PRINT cast(@BinaryVariable as real)
END
GO
EXEC p453 1.1;
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: cannot cast type varbinary to real)~~


DROP PROCEDURE p453;
DROP TABLE t453_real;
DROP TABLE t453_dp;
GO
